iOSNativeLoginTemplate/iOSNativeLoginTemplate/NativeLoginView.swift

```diff
@@ -31,134 +31,183 @@ import SalesforceSDKCore
+                                case .invalidCredentials:
+                                    errorMessage = "Please check your username and password."
+                                    break
+                                case .invalidUsername:
+                                    errorMessage = "Username format is incorrect."
+                                    break
+                                case .invalidPassword:
+                                    errorMessage = "Invalid password."
+                                    break
+                                case .unknownError:
+                                    errorMessage = "An unknown error has occurred."
+                                    break
+                                case .success:
+                                    self.password = ""
+                                }
                             }
+                        } label: {
+                            HStack {
+                                Image(systemName: "lock")
+                                Text("Log In")
+                            }.frame(minWidth: 150)
                         }
-                    } label: {
-                        HStack {
-                            Image(systemName: "lock")
-                            Text("Log In")
-                        }.frame(minWidth: 150)
-                    }
-                    .buttonStyle(.bordered)
-                    .tint(.blue)
-                    .zIndex(2.0)
-                }.padding(.bottom, 0)
-            }
-            .frame(maxHeight: .infinity, alignment: .center)
-            .padding(.bottom, 125)
-            
-            // Fallback to webview based authentication.
-            Button("Looking for Salesforce Log In?") {
-                SalesforceManager.shared.nativeLoginManager().fallbackToWebAuthentication()
+                        .buttonStyle(.bordered)
+                        .tint(.blue)
+                        .zIndex(2.0)
+                    }.padding(.bottom, 0)
+                }
+                
+                // Other login options.
+                Button("Need to register, reset your password or login without a password?") {
+                    navigationPathObservable.navigationPath.append(.NativeRequestOtpView)
+                }
+                
+                Spacer()
+                
+                // Fallback to webview based authentication.
+                Button("Looking for Salesforce Log In?") {
+                    SalesforceManager.shared.nativeLoginManager().fallbackToWebAuthentication()
+                }
             }
-        }.background(Gradient(colors: [.blue, .cyan, .green]).opacity(0.6))
+            .background(
+                Gradient(colors: [.blue, .cyan, .green]).opacity(0.6)
+            )
             .blur(radius: self.isAuthenticating ? 2.0 : 0.0)
+            .navigationDestination(for: NavigationDestination.self) { navigationDestination in
```

Comment:
Here's the new `navigationDestination` modifier for the `NavigationStack`.  This declarative navigation DSL is actually a lot of fun (Though Xcode's SwiftUI "Canvas" crashes _a lot_.)